prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -270,11 +270,12 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
          *
          * @return a [Pair<Message, HL7MessageParseAndConvertConfiguration?>] with parsed message and optional type
          */
-        fun parseHL7Message(rawHL7: String): Pair<Message, HL7MessageParseAndConvertConfiguration?> {
+        fun parseHL7Message(
+            rawHL7: String,
+            parseConfiguration: HL7MessageParseAndConvertConfiguration?,
+        ): Message {
             val hl7MessageType = getMessageType(rawHL7)
-            val parseConfiguration = messageToConfigMap[hl7MessageType]
-            val message = getHL7ParsingContext(hl7MessageType, parseConfiguration).pipeParser.parse(rawHL7)
-            return Pair(message, parseConfiguration)
+            return getHL7ParsingContext(hl7MessageType, parseConfiguration).pipeParser.parse(rawHL7)
```

Comment:
I think we need to look into `getHL7ParsingContext` and evaluate the business logic there and align with product. I'll think on it and maybe create a ticket, but I think this is fine for now if its well tested.

Reply:
Yep... I think we have it isolated well now so that we make tactical changes here